components/src/core/Drawer/DrawerNavItem.tsx

```diff
@@ -196,7 +206,12 @@ export const DrawerNavItem: React.FC<DrawerNavItem> = (props) => {
     const active = activeItem === itemID;
     const infoListItemClasses = {
         root: defaultClasses.infoListItem,
-        title: depth > 0 ? clsx(defaultClasses.nestedTitle, classes.nestedTitle) : '',
+        title: clsx(
+            depth > 0 && defaultClasses.nestedTitle,
+            depth > 0 && classes.nestedTitle,
+            hidePadding && !icon && defaultClasses.noIconTitle
+        ),
+        subtitle: clsx(hidePadding && !icon && defaultClasses.noIconTitle),
```

Comment:
clsx has a nice syntax for conditionally including classes:
```suggestion
        title: clsx({
            [defaultClasses.nestedTitle]: depth >0,
            [classes.nestedTitle]: depth > 0,
            [defaultClasses.noIconTitle]: hidePadding && !icon,
        }),
        subtitle: clsx({
            [defaultClasses.noIconTitle]: hidePadding && !icon,
        }),
```